src/main/java/seedu/duke/parser/commandparser/UpdateParser.java

```diff
@@ -1,11 +1,215 @@
+import seedu.duke.commands.updates.UpdateMemberCommand;
+import seedu.duke.commands.updates.UpdateTaskCommand;
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.exceptions.parserexceptions.InvalidBudgetException;
+import seedu.duke.items.Event;
+import seedu.duke.items.Task;
+import seedu.duke.items.characteristics.Member;
 import seedu.duke.parser.Parser;
 
+import java.time.LocalDateTime;
+
 public abstract class UpdateParser extends Parser {
 
-    public Command getUpdateCommand(String commandDetails) {
+    private static final int INDEX_OF_TITLE = 0;
+    private static final int INDEX_OF_VENUE = 2;
+    private static final int INDEX_OF_DESCRIPTION = 1;
+    protected static final String TITLE_FLAG = "title/";
+    protected static final String DATE_FLAG = "date/";
+    protected static final String DEADLINE_FLAG = "deadline/";
+    protected static final String VENUE_FLAG = "venue/";
+    protected static final String BUDGET_FLAG = "budget/";
+    protected static final String DESCRIPTION_FLAG = "description/";
+    protected static final String TASK_FLAG = "task/";
+    protected static final String MEMBER_FLAG = "member/";
+    protected static final String REMOVE_FLAG = "remove/";
+    protected static final String ADD_FLAG = "add";
+
+    public static Command getUpdateCommand(String commandDetails) {
+        try {
+            int eventIndex = Integer.parseInt(commandDetails) - 1;
+            Event eventToBeUpdated = Duke.eventCatalog.get(eventIndex);
+            Ui.printUpdateEventDetails(eventToBeUpdated);
+            Ui.updateIntroMessage();
+            String userInput = Ui.readInput();
+            String[] userUpdates = userInput.trim().split(">+");
+            Command result = parseUpdateEvent(userUpdates, eventToBeUpdated);
+            Ui.printLineBreak();
+            return result;
+
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid index of event, please type an integer!");
+        } catch (InvalidBudgetException | DukeException e) {
+            System.out.println(e.getMessage());
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Please key in your updates!");
+        } catch (IndexOutOfBoundsException e) {
+            Ui.printLineBreak();
+            System.out.println("That selection does not exist!");
+        }
         return null;
     }
-}
+
+    private static Command parseUpdateEvent(String[] userUpdates, Event event)
+            throws DukeException, InvalidBudgetException {
+        String[] parsedAttributes = new String[3];
+        LocalDateTime dateTime = null;
+        double budget = 0;
+        boolean isFlag;
+        for (String update : userUpdates) {
+            isFlag = false;
+            String[] attribute = update.trim().split("/+");
+            if (update.contains(TITLE_FLAG)) {
+                isFlag = true;
+                parsedAttributes[INDEX_OF_TITLE] = attribute[1];
+            }
```

Comment:
On inspection, the current method of identifying flags might lead to inaccurate parsing of user input. For instance, a user might input `adsftitle/asdf` and the conditional check in line 68 will still pass since it looks at the entire string of user input and does a simple check for whether it contains that flag (i.e. position of flag relative to the entire string is not important in the check). Perhaps someone can help verify this as well.

A possible solution would be to use `update.startsWith(XYZ_FLAG)` instead. However, this might require additional methods like `trim()` to be performed on each `update` string first.

It would also be to use the `retrieveItemAttribute()` method from the inherited `Parser` class. This would further unify and standardize the parsing logic for the program in retrieving attributes and also standardize the flags used. That being said, the `retrieveItemAttribute()` method has to also be updated as the current implementation of it has the same issue as well.